codegen/src/main/kotlin/tools/samt/codegen/kotlin/KotlinTypesGenerator.kt

```diff
@@ -0,0 +1,139 @@
+package tools.samt.codegen.kotlin
+
+import tools.samt.api.plugin.CodegenFile
+import tools.samt.api.plugin.Generator
+import tools.samt.api.plugin.GeneratorParams
+import tools.samt.api.types.*
+
+object KotlinTypesGenerator : Generator {
+    override val name: String = "kotlin-types"
+    override fun generate(generatorParams: GeneratorParams): List<CodegenFile> {
+        generatorParams.packages.forEach {
+            generatePackage(it, generatorParams.options)
+        }
+        val result = emittedFiles.toList()
+        emittedFiles.clear()
+        return result
+    }
+
+    private val emittedFiles = mutableListOf<CodegenFile>()
+
+    private fun generatePackage(pack: SamtPackage, options: Map<String, String>) {
+        if (pack.hasModelTypes()) {
+            val packageSource = buildString {
+                appendLine(GeneratedFilePreamble)
+                appendLine()
+                appendLine("package ${pack.getQualifiedName(options)}")
+                appendLine()
+
+                pack.records.forEach {
+                    appendRecord(it, options)
+                }
+
+                pack.enums.forEach {
+                    appendEnum(it)
+                }
+
+                pack.aliases.forEach {
+                    appendAlias(it, options)
+                }
+
+                pack.services.forEach {
+                    appendService(it, options)
+                }
+            }
+
+            val filePath = "${pack.getQualifiedName(options).replace('.', '/')}/Types.kt"
+            val file = CodegenFile(filePath, packageSource)
+            emittedFiles.add(file)
+        }
+    }
+
+    private fun StringBuilder.appendRecord(record: RecordType, options: Map<String, String>) {
+        if (record.fields.isEmpty()) {
+            appendLine("class ${record.name}")
+            appendLine()
+            return
+        }
+
+        appendLine("data class ${record.name}(")
+        record.fields.forEach { field ->
+            val fullyQualifiedName = field.type.getQualifiedName(options)
+
+            if (field.type.isRuntimeOptional) {
+                appendLine("    val ${field.name}: $fullyQualifiedName = null,")
+            } else {
+                appendLine("    val ${field.name}: $fullyQualifiedName,")
+            }
+        }
+        appendLine(")")
+        appendLine()
+    }
+
+    private fun StringBuilder.appendEnum(enum: EnumType) {
+        appendLine("enum class ${enum.name} {")
+        appendLine("    /** Default value used when the enum could not be parsed */")
+        appendLine("    FAILED_TO_PARSE,")
```

Comment:
maybe make it `failed to parse` like in other places to avoid conflicts.

Reply:
Enum entries should not contain whitespaces, so I would leave it as-is for the time being